test/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware.Tests/Buffering/HttpRequestBufferLoggerBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if NET9_0_OR_GREATER
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Diagnostics.Logging.Test;
+
+public class HttpRequestBufferLoggerBuilderExtensionsTests
+{
+    [Fact]
+    public void AddHttpRequestBuffer_RegistersInDI()
+    {
+        var serviceCollection = new ServiceCollection();
+        serviceCollection.AddLogging(builder =>
+        {
+            builder.AddHttpRequestBuffer(LogLevel.Warning);
+        });
+
+        var serviceProvider = serviceCollection.BuildServiceProvider();
+        var buffer = serviceProvider.GetService<ILoggingBuffer>();
```

Comment:
Are we expecting users to query the buffer from the DI container for some reason? I wasn't expecting them to do that. If there was going to be some buffer in the DI container I also would have expected it to be the GlobalBuffer as there is no http request in progress.

Reply:
Modified test to make it as much canonical as possible in the circumstances  of deterministic unit tests.